src/Compilers/CSharp/Portable/Binder/AwaitableInfo.cs

```diff
@@ -0,0 +1,53 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using Microsoft.CodeAnalysis.CSharp.Symbols;
+using Roslyn.Utilities;
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    /// <summary>
+    /// Internal structure containing all semantic information about an await expression.
+    /// </summary>
+    internal class AwaitableInfo : IEquatable<AwaitableInfo>
+    {
+        public MethodSymbol GetAwaiter { get; }
+
+        public PropertySymbol IsCompleted { get; }
+
+        public MethodSymbol GetResult { get; }
+
+        public TypeSymbol Type { get; }
+
+        public bool IsDynamic => GetResult == null;
+
+        internal AwaitableInfo(MethodSymbol getAwaiterMethod,
+            PropertySymbol isCompletedProperty,
+            MethodSymbol getResultMethod,
+            TypeSymbol type)
+        {
+            this.GetAwaiter = getAwaiterMethod;
+            this.IsCompleted = isCompletedProperty;
+            this.GetResult = getResultMethod;
+            this.Type = type;
+        }
+
+        public override bool Equals(object obj)
+        {
+            return obj is AwaitableInfo && Equals((AwaitableInfo)obj);
```

Comment:
>obj is AwaitableInfo && Equals((AwaitableInfo)obj) [](start = 19, length = 50)

```Equals(obj as AwaitableInfo)```? Since AwaitableInfo is a class.